File: climbing-the-leaderboard.py
leaderboard = [100, 100, 50, 40, 40, 20, 10];
game = [5, 25, 50, 120];

def climbingLeaderboard (ranked, player):
    ranked = sorted(list(set(ranked)), reverse=True)
    player = sorted(player, reverse=True);

    res = []
    j = 0
    l = len(ranked);

    for i in range(len(player)):
        while j < l and player[i] < ranked[j]:
            j+=1
        
        res.append(j+1)
    print(res)
    return res[::-1]

climbingLeaderboard(leaderboard, game)

File: climbing-the-leaderboard.js
'use strict';

const leaderboard = [100, 100, 50, 40, 40, 20, 10];
const game = [5, 25, 50, 120];

const climbingLeaderboard = (ranked, player) => {
  // remover dupes y ordenar asc.
  const rankedSort = [...new Set(ranked)].sort((a, b) => a - b);
  player = player.sort((a, b) => a - b);

  const res = [];
  let j = 0;
  const l = rankedSort.length;

  console.log(rankedSort);
  console.log(player);

  for (const i of player) {
    while (j < l && player[i] <  rankedSort[j] ) { 
      j++;
      res.push(j + 1)
    }
  }
  return res;

}

console.log(climbingLeaderboard(leaderboard, game))


